snapshots/snapshotter.go

```diff
@@ -315,8 +315,11 @@ type Snapshotter interface {
 	Remove(ctx context.Context, key string) error
 
 	// Walk all snapshots in the snapshotter. For each snapshot in the
-	// snapshotter, the function will be called.
-	Walk(ctx context.Context, fn func(context.Context, Info) error) error
+	// snapshotter, the function will be called, unless filters were used.
+	// Zero or more filters may be provided as strings. Only events that match
+	// *any* of the provided filters will be sent on the channel. The filters use
+	// the standard containerd filters package syntax.
+	Walk(ctx context.Context, fn func(context.Context, Info) error, filters ...string) error
```

Comment:
Do we really need to change the interface? Wondering this could be implemented as a custom `fn func(context.Context, Info) error`

Reply:
The problem with a custom func is that it gives no performance advantage. The filters allow the backend to limit what gets sent back as well as the possibility of using indexing. If we don't fix it now, it will just get harder to fix in the future. Adding any new function would be the same issue.